src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -518,7 +525,7 @@ private bool CheckIfUnsolicitedFutureHeader(List<BlockHeader> headers)
         /// <summary>Sync when handshake is finished.</summary>
         private async Task OnStateChangedAsync(INetworkPeer peer, NetworkPeerState oldState)
         {
-            if (peer.State == NetworkPeerState.HandShaked)
+            if (peer.State == NetworkPeerState.HandShaked && this.cachedHeaders.Count == 0)
```

Comment:
This change is not needed. When state becomes handshaked we have no cached headers.

Reply:
Will revert 👍